src/test/miner_tests.cpp

```diff
@@ -158,11 +160,13 @@ void MinerTestingSetup::TestPackageSelection(const CScript& scriptPubKey, const
     tx.vout[0].nValue = 5000000000LL - 1000 - 50000 - feeToUse;
     Txid hashLowFeeTx = tx.GetHash();
     AddToMempool(tx_mempool, entry.Fee(feeToUse).FromTx(tx));
-    pblocktemplate = AssemblerForTest(tx_mempool, options).CreateNewBlock();
+    block_template = mining->createNewBlock(options);
+    BOOST_REQUIRE(block_template);
+    block = block_template->getBlock();
```

Comment:
We repeat this pattern throughout the file.
Unless I'm missing something, seems like a good opportunity to de-duplicate with something like:

```c++
CBlock CreateAndGetBlock(const node::BlockCreateOptions& options)
{
    auto mining{MakeMining()};
    BOOST_REQUIRE(mining);
    std::unique_ptr<BlockTemplate> block_template = mining->createNewBlock(options);
    BOOST_REQUIRE(block_template);
    return block_template->getBlock();
}
```

Reply:
That makes sense. Will do if I need to retouch.

Reply:
I briefly tried this with #31283 but it's not very convenient. Sometimes I need the `CBlock`, but other times I need the `BlockTemplate`, still other times I need the template to be `nullptr`.